src/rotation.jl

```diff
@@ -0,0 +1,288 @@
+        (isodd(size(arr, 1)) && isodd(size(arr, 2)) && 
+        (rotation_center[1] == size(arr, 1) ÷ 2 + 1 && rotation_center[1] == size(arr, 2) ÷ 2 + 1))
+        if θ ≈ π / 2 
+            if adjoint == false
+                out .= reverse(PermutedDimsArray(arr, (2, 1, 3, 4)), dims=(2,))
+            else
+                out .= reverse(PermutedDimsArray(arr, (2, 1, 3, 4)), dims=(1,))
+            end
+            return true
+        elseif θ ≈ π
+            out .= reverse(arr, dims=(1,2))
+            return true
+        elseif θ ≈ 3 / 2 * π
+            if adjoint == false
+                out .= reverse(PermutedDimsArray(arr, (2, 1, 3, 4)), dims=(1,))
+            else
+                out .= reverse(PermutedDimsArray(arr, (2, 1, 3, 4)), dims=(2,))
+            end
+            return true
+        end
+    end
+
+    return false
+end
+
+
+"""
+    imrotate(arr::AbstractArray{T, 4}, θ; method=:bilinear, rotation_center=size(arr) .÷ 2 .+ 1)
+
+Rotates a matrix around the center pixel `rotation_center`. `rotation_center` is defined such that there
+is a real center pixel for even and odd values which is rotated around.
+The angle `θ` is interpreted in radians.
+
+The adjoint is defined with ChainRulesCore.jl. This method also runs with CUDA (and in principle all KernelAbstractions.jl supported backends).
+
+# Keywords
+* `method=:bilinear` for bilinear interpolation or `method=:nearest` for nearest neighbour
+* `rotation_center=size(arr) .÷ 2 .+ 1` means there is a real center pixel around it is rotated.
+
+# Examples
+```julia-repl
+julia> arr = zeros((4,4,1,1)); arr[2,2,1,1] = 1;
+
+julia> arr
+4×4×1×1 Array{Float64, 4}:
+[:, :, 1, 1] =
+ 0.0  0.0  0.0  0.0
+ 0.0  1.0  0.0  0.0
+ 0.0  0.0  0.0  0.0
+ 0.0  0.0  0.0  0.0
+
+julia> NNlib.imrotate(arr, deg2rad(90)) # rotation around (3,3)
+4×4×1×1 Array{Float64, 4}:
+[:, :, 1, 1] =
+ 0.0  0.0  0.0  0.0
+ 0.0  0.0  0.0  1.0
+ 0.0  0.0  0.0  0.0
+ 0.0  0.0  0.0  0.0
+
+julia> NNlib.imrotate(arr, deg2rad(90), rotation_center=(2,2))
+4×4×1×1 Array{Float64, 4}:
+[:, :, 1, 1] =
+ 0.0  0.0  0.0  0.0
+ 0.0  1.0  0.0  0.0
+ 0.0  0.0  0.0  0.0
+ 0.0  0.0  0.0  0.0
+
+julia> arr = zeros((3,3,1,1)); arr[1,2,1,1] = 1
+1
+
+julia> arr
+3×3×1×1 Array{Float64, 4}:
+[:, :, 1, 1] =
+ 0.0  1.0  0.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+
+julia> NNlib.imrotate(arr, deg2rad(45))
+3×3×1×1 Array{Float64, 4}:
+[:, :, 1, 1] =
+ 0.0  0.207107  0.0
+ 0.0  0.0       0.207107
+ 0.0  0.0       0.0
+
+julia> NNlib.imrotate(arr, deg2rad(45), method=:nearest)
+3×3×1×1 Array{Float64, 4}:
+[:, :, 1, 1] =
+ 0.0  0.0  1.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+```
+"""
+function imrotate(arr::AbstractArray{T, 4}, θ; method=:bilinear, rotation_center=size(arr) .÷ 2 .+ 1) where T
+    @assert (T <: Integer && method==:nearest || !(T <: Integer)) "If the array has an Int eltype, only method=:nearest is supported"
+    @assert typeof(rotation_center) <: Tuple "rotation_center keyword has to be a tuple"
```

Comment:
This is minor but I think the first should test and `throw`, and the second should just be `rotation_center::Tuple` in the signature.